binheap.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+class Heap(object):
+    """Create a Heap."""
+
+    def __init__(self, iterable=None):
+        """Initialize Heap."""
+        # if iterable:
+        #     self.container = list(iterable)
+        #     for i in self.container:
+        #         self.push(i)
+        # else:
+        self.container = []
+        self.current_size = 0
+
+    # def swap(self, value):
+    #     """Swap parent and child."""
+    #     val_index = self.container.index(value)
+    #     current_parent = self.parent(val_index)
+    #     self.container[current_parent] = value
+    #     self.container[val_index] = current_parent
+    #     return self.container[current_parent]
+
+    def push(self, value):
+        """Push a value to the Heap and maintain Heap structure."""
+        self.container.append(value)
+        child = len(self.container) - 1
+        parent = (child - 1) // 2
+        while child > 1 and self.container[child - 1] > self.container[parent - 1]:
```

Comment:
For this class, make sure to adhere to the 80 character line limit. We're mostly on laptops, so screen space is at a premium, and it's part of the linting standards. This is one way to make it work:

```python
        while (child > 1 and
               self.container[child - 1] > self.container[parent - 1]):
```

See [implicit line joining](https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length)